build.yaml

```diff
@@ -831,10 +833,15 @@ common_cli:
   upload_new_docs_html_to_s3_staging_docs: &upload_new_docs_html_to_s3_staging_docs |
     # uploads the new generated html to a folder on the S3 staging-docs bucket
     cd _build/html
-    aws --region ${S3_ENDPOINT} s3 sync . s3://clusterhq-staging-docs/$TRIGGERED_BRANCH
+    aws --region ${S3_DOCS_REGION} s3 sync . s3://clusterhq-staging-docs/$TRIGGERED_BRANCH
     echo "This branch is available at :"
     echo "http://clusterhq-staging-docs.s3.amazonaws.com/$TRIGGERED_BRANCH/index.html"
 
+  sync_master_docs_to_doc_dev: &sync_master_docs_to_doc_dev |
+    # ensure master docs at doc-dev are up-to-date
+    aws --region ${S3_DOCS_REGION} s3 sync --acl public-read \
```

Comment:
My preference would be for this to only run on commits to master rather than having this in the job configuration for every branch. The `run_sphinx` job is parameterised with the parameter `TRIGGERED_BRANCH`. We could use this to make this step conditional like you previously mentioned.